sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveInspectors.scala

```diff
@@ -847,6 +847,8 @@ private[hive] trait HiveInspectors {
 
         ObjectInspectorFactory.getStandardConstantMapObjectInspector(keyOI, valueOI, jmap)
       }
+    case Literal(_, dt: StructType) =>
+      toInspector(dt)
```

Comment:
Can we have a test case for this, @cxzl25 ? Otherwise, we cannot protect your PR from the future regression.

Reply:
ok, add a new test, please help review again, thank you